### PR TITLE
Flexi solc

### PR DESCRIPTION
**Pull request by gsalzer**
This merge adds support for arbitrary Solidity versions.

### CODE REVIEW

File: src/output_parser/sarif_vulnerability_mapping.csv [hunk 1]
```diff
@@ -53,7 +53,8 @@ oyente,Arithmetic_12,Integer Underflow.,Arithmetic
 oyente,AccessControl_16,Parity Multisig Bug 2.,AccessControl
 oyente,Reentrancy_6,Re-Entrancy Vulnerability.,Reentrancy
 oyente,TimeManipulation_3,Timestamp Dependency.,TimeManipulation
-securify,Reentrancy_7,DAO,Reentrancy
+oyente,FrontRunning_6,Transaction-Ordering Dependency.,FrontRunning
+Dsecurify,Reentrancy_7,DAO,Reentrancy
```
Comment (tdurieux) [inline src/output_parser/sarif_vulnerability_mapping.csv:57]: I think there is a D at the beginning of the line that should not be there

Comment (gsalzer) [reply src/output_parser/sarif_vulnerability_mapping.csv]: Well observed, apparently a character that went to the wrong window. I pushed the fix. Moreover, I added another test exercising the various flag, see `tests/coverage.sh`. Moreover, there were still some path inconsistencies regarding whether the pathes are relative to the current dir or to smartbugs' home dir. As it is now, it is possible to install smartbugs globally, and all files written by smartbugs are put into the user space (either below the current working dir or into `$HOME/.solcx`).

Please recheck again ...

File: src/config.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+def is_remote_info(info):
+    return isinstance(info,dict) and 'url' in info and 'local_dir' in info
+
+def get_remote_info(info):
+    return (info['url'], info['local_dir'], info['subsets'])
+
+import pandas
+VULNERABILITY_MAP_PATH = os.path.join(BASE_DIR, 'src', 'output_parser', 'sarif_vulnerability_mapping.csv')
+VULNERABILITY_MAP = pandas.read_csv(VULNERABILITY_MAP_PATH)
```
Comment (tdurieux) [inline src/config.py:41]: pandas dep should not be imported?

Comment (tdurieux) [reply src/config.py:41]: When this is solved I think it can be merged

Comment (gsalzer) [reply src/config.py:41]: Using pandas to read a csv file is like cracking a nut with a sledge-hammer. But this is the way how Smartbugs does it so far, not my idea, see https://github.com/smartbugs/smartbugs/blob/d6f70ad71610d2af27a9439d1fb8166c75ed548e/src/output_parser/SarifHolder.py .
My first impulse was to replace it, but then postponed it as it is a code change functionally not necessary. But it hurts, I agree, and will replace it.

Comment (tdurieux) [reply src/config.py:41]: I meant that `import pandas` was missing but I did not see that it was in line 39. Sorry.

I don't think we need to replace it right now. Lets focus on something that has more value for the project.

**Review by tdurieux — APPROVED** [review]

**Pull request merged**